File: src/main/java/kanti/sl/arguments/StateArgumentsImpl.java
package kanti.sl.arguments;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.*;

public class StateArgumentsImpl implements StateArguments {

	private final Map<String, StateArgument> argsMap;

	public StateArgumentsImpl(@NotNull StateArgument... args) {
		argsMap = new LinkedHashMap<>();
		for (StateArgument arg : args) {
 			argsMap.put(arg.getKey(), arg);
		}
	}

	@Override
	@NotNull
	public List<StateArgument> getArguments() {
		return new ArrayList<>(argsMap.values());
	}

	@Nullable
	@Override
	public StateArgument get(@NotNull String key) {
		return argsMap.get(key);
	}

}

File: src/main/kotlin/kanti/sl/arguments/StateArgumentsImpl.kt
package kanti.sl.arguments

class StateArgumentsImpl(
	arguments: Iterable<StateArgument>
) : StateArguments {

	private val statesMap: Map<String, StateArgument>

	override val arguments: List<StateArgument>
		get() = statesMap.values.toList()

	init {
		statesMap = buildMap {
			for (arg in arguments) {
				put(arg.name, arg)
			}
		}
	}

	override operator fun get(key: String): StateArgument? {
		return statesMap[key]
	}

}